src/server/req_delete.c

```diff
@@ -201,103 +201,169 @@ acct_del_write(char *jid, job *pjob, struct batch_request *preq, int nomail)
  *		done by req_deletejob()
  *
  * @param[in]	preq	- Batch request structure.
+ * @param[in]	pjob	- pointer to job
+ * @param[in]	parent	- pointer to parent job for sub job
+ * @param[in]	jt	- job type
  *
  * @return	int
  * @retval	TRUE	- Job history  has been purged
  * @retval	FALSE	- Job is not a history job
  */
 int
-check_deletehistoryjob(struct batch_request * preq)
+check_deletehistoryjob(struct batch_request * preq, job *pjob, job *parent, int jt)
 {
-	job *histpjob;
-	job *pjob;
-	int historyjob;
-	int histerr;
 	char *jid;
+	job *psjob;
+	char *sjid;
+	char *pc;
+	int i,j;
+	int x, y, z;
+
 	jid = preq->rq_ind.rq_delete.rq_objname;
 
-	histpjob = find_job(jid);
-	/*
-	 * if histpjob returns NULL then we can be sure its a array subjob or a range of subjobs . If the array
-	 * subjob or range of subjobs are in a history state then reject the request as we cant delete history
-	 * of array subjobs
-	 */
-	if (histpjob == NULL) {
-		pjob = find_arrayparent(jid);
-		if ((histerr = svr_chk_histjob(pjob))) {
-			req_reject(PBSE_NOHISTARRAYSUBJOB, 0, preq);
-			return TRUE;
-		} else {
-			/*
-			 * Job is in a Non Finished state . It must be terminated and then its history
-			 *  should be purged .
-			 */
-			return FALSE;
-		}
+	if (PBSE_HISTJOBID != svr_chk_histjob(pjob))
+		return FALSE;
+
+
+	/* Issue history job delete request to remote server if job is moved. */
+	if (pjob->ji_qs.ji_state == JOB_STATE_MOVED)
+		issue_delete(pjob, preq);
+
+	if ((jt == IS_ARRAY_NO) || (jt == IS_ARRAY_Single)) {
+		snprintf(log_buffer, sizeof(log_buffer),
+			msg_job_history_delete, preq->rq_user,
+			preq->rq_host);
+		log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_INFO,
+			jid,
+			log_buffer);
+
+
+		job_purge(pjob);
+		preq->rq_reply.brp_code = PBSE_HISTJOBDELETED;
+		reply_send(preq);
+		return TRUE;
 	}
 
-	historyjob = svr_chk_histjob(histpjob);
-	if (historyjob == PBSE_HISTJOBID) {
+	/* handle Arrayjob */
+	if (jt == IS_ARRAY_ArrayJob) {
+
+		/* The Array Job itself ... */
+		/* for each subjob of array do purge */
+
+		if (pjob->ji_ajtrk) {
+			for (i = 0; i < pjob->ji_ajtrk->tkm_ct; i++) {
+				sjid = mk_subjob_id(pjob, i);
+
+				psjob = find_job(sjid);
+
+				if (psjob) {
+					snprintf(log_buffer, sizeof(log_buffer),
+						msg_job_history_delete, preq->rq_user,
+						preq->rq_host);
+					log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB, LOG_INFO,
+							sjid,
+						log_buffer);
```

Comment:
Align sjid, log_buffer in the same line for better readability.

Reply:
fixed in rc 5